packages/mdc-menu-surface/foundation.js

```diff
@@ -360,28 +383,29 @@ class MDCMenuSurfaceFoundation extends MDCFoundation {
 
     const corner = this.getOriginCorner_();
     const maxMenuSurfaceHeight = this.getMenuSurfaceMaxHeight_(corner);
-    let verticalAlignment = (corner & MenuSurfaceCornerBit.BOTTOM) ? 'bottom' : 'top';
+    const verticalAlignment = (corner & MenuSurfaceCornerBit.BOTTOM) ? 'bottom' : 'top';
     let horizontalAlignment = (corner & MenuSurfaceCornerBit.RIGHT) ? 'right' : 'left';
     const horizontalOffset = this.getHorizontalOriginOffset_(corner);
     const verticalOffset = this.getVerticalOriginOffset_(corner);
-    const position = {
-      [horizontalAlignment]: horizontalOffset ? horizontalOffset + 'px' : '0',
-      [verticalAlignment]: verticalOffset ? verticalOffset + 'px' : '0',
+    let position = {
+      [horizontalAlignment]: horizontalOffset ? horizontalOffset : '0',
+      [verticalAlignment]: verticalOffset ? verticalOffset : '0',
     };
-    const {anchorWidth, surfaceHeight, surfaceWidth} = this.measures_;
+    const {anchorWidth, surfaceWidth} = this.measures_;
     // Center align when anchor width is comparable or greater than menu surface, otherwise keep corner.
     if (anchorWidth / surfaceWidth > numbers.ANCHOR_TO_MENU_SURFACE_WIDTH_RATIO) {
       horizontalAlignment = 'center';
     }
 
-    // Adjust vertical origin when menu surface is positioned with significant offset from anchor. This is done so that
-    // scale animation is "anchored" on the anchor.
-    if (!(this.anchorCorner_ & MenuSurfaceCornerBit.BOTTOM) &&
-        Math.abs(verticalOffset / surfaceHeight) > numbers.OFFSET_TO_MENU_SURFACE_HEIGHT_RATIO) {
-      const verticalOffsetPercent = Math.abs(verticalOffset / surfaceHeight) * 100;
-      const originPercent = (corner & MenuSurfaceCornerBit.BOTTOM)
-        ? 100 - verticalOffsetPercent : verticalOffsetPercent;
-      verticalAlignment = Math.round(originPercent * 100) / 100 + '%';
+    // If the menu-surface has been hoisted to the body, it's no longer relative to the anchor element
+    if (this.hoistedElement_) {
+      position = this.adjustPositionForHoistedElement_(position);
+    }
+
+    for (const prop in position) {
+      if (position.hasOwnProperty(prop) && position[prop] !== '0') {
+        position[prop] = `${parseInt(position[prop], 0)}px`;
```

Comment:
You get decimals in horizontal and vertical alignment?

Reply:
This should be `10`

Reply:
The 0 should be 10?